### PR TITLE
Add blasterswap_blast_base_trades

### PR DESCRIPTION
**Pull request by Hosuke**
### Contribution type
Please check the type of contribution this pull request is for:

- [x] New spell(s)
- [ ] Adding to existing spell lineage
- [ ] Bug fix

**Note:** You can safely discard any section below which doesn't apply based on selection above

---

### For new spell(s)
If you are building new spell(s), please provide the following information:

- **Spell name(s):** *schema.table_name*
- **Description:** [Detailed description of the new spell(s) and their purpose]
- **Who are the new spell(s) for?** [Internal team or general community]
- **How will the new spell(s) be used downstream?** [Description of downstream usage]
- **Implementation details:** [Information on how the spell(s) are implemented]
- **Test instructions:** [How to test the new spell(s)]
- **Related issue(s):** [Link to related issues, if any]

---

### For adding to existing spell lineage
If you are adding to an existing spell lineage, please provide the following information:

- **Description:** [Description of the changes made]

---

### For bug fixes
If you are fixing a bug, please provide the following information:

- **Description:** [Description of the bug fix]
- **Steps to reproduce:** [How to reproduce the bug]
- **Implementation details:** [Information on how the bug was fixed]
- **Test instructions:** [How to test the fix]
- **Related issue(s):** [Link to related issues, if any]

---

### Additional information
Please provide any additional information that might help us review your pull request:

- [Any additional information]

---

Thank you for your contribution!

### CODE REVIEW

File: dex/models/trades/blast/platforms/blasterswap_blast_base_trades.sql [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+WITH
+
+unioned_evt_sources AS (
+    {% for evt_source in evt_sources %}
+        SELECT
+            *
+        FROM
+            {{ evt_source }}
+        {% if not loop.last %}
```
Comment (Hosuke) [inline dex/models/trades/blast/platforms/blasterswap_blast_base_trades.sql:28]:
```suggestion
        FROM
            {{ evt_source }}
        {% if is_incremental() %}
        WHERE {{ incremental_predicate('evt_block_time') }}
        {% endif %}
        {% if not loop.last %}
```

File: dex/models/trades/blast/platforms/blasterswap_blast_base_trades.sql [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        alias = 'base_trades',
+        materialized = 'incremental',
+        file_format = 'delta',
+        incremental_strategy = 'merge',
+        unique_key = ['tx_hash', 'evt_index'],
+        incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_time')]
+    )
+}}
+
+{% set evt_sources = [
```
Comment (aalan3) [inline dex/models/trades/blast/platforms/blasterswap_blast_base_trades.sql:13]: Is this something specific to balsterswap, do they only support these coins or are we missing decoding for Pair generally_

Comment (Hosuke) [reply dex/models/trades/blast/platforms/blasterswap_blast_base_trades.sql:13]: We may missing decoding for Pair generally.
But blasterswap v2 is only limited to these 4 pools:
https://github.com/blasterswap/blasterswap-core-v2?tab=readme-ov-file#pairs

**Review by jeff-dude — APPROVED** [review]

File: dex/models/trades/blast/platforms/blasterswap_blast_base_trades.sql (hunk 2 above)
Comment (jeff-dude) [reply dex/models/trades/blast/platforms/blasterswap_blast_base_trades.sql:13]: nice find

**Pull request merged**